### PR TITLE
[TA] Make fromString method package-private

### PR DESCRIPTION
**Pull request by mssfang**
fixes: #11213

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java [hunk 1]
```diff
@@ -144,43 +141,21 @@
      * @return The {@link AnalyzeSentimentResult} to be returned by the SDK.
      */
     private AnalyzeSentimentResult convertToAnalyzeSentimentResult(DocumentSentiment documentSentiment) {
-        // Document text sentiment
-        final TextSentiment documentSentimentLabel = TextSentiment.fromString(
-            documentSentiment.getSentiment().toString());
-        if (documentSentimentLabel == null) {
```
Comment (mssfang) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java:150]: [ERROR] Redundant nullcheck of documentSentimentLabel, which is known to be non-null in com.azure.ai.textanalytics.AnalyzeSentimentAsyncClient.convertToAnalyzeSentimentResult(DocumentSentiment) [com.azure.ai.textanalytics.AnalyzeSentimentAsyncClient] Redundant null check at AnalyzeSentimentAsyncClient.java:[line 147] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[ERROR] Redundant nullcheck of sentenceSentimentLabel, which is known to be non-null in com.azure.ai.textanalytics.AnalyzeSentimentAsyncClient.lambda$convertToAnalyzeSentimentResult$9(SentenceSentiment) [com.azure.ai.textanalytics.AnalyzeSentimentAsyncClient] Redundant null check at AnalyzeSentimentAsyncClient.java:[line 161] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/WarningCode.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
      * @return The corresponding {@link WarningCode}.
      */
     @JsonCreator
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/WarningCode.java:32]: don't need this>

Comment (samvaity) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/WarningCode.java:32]: Is this a generated model https://github.com/Azure/azure-sdk-for-java/pull/11227/files#diff-2683c3cab1fd268997bc5df017c80afcL3?

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java (hunk 1 above)
Comment (samvaity) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java:150]: This started coming up from the current change?

Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java:150]: Yes. Null checking here is an redundant check. As the error message explained here.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/WarningCode.java (hunk 2 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/WarningCode.java:32]: WarningCode is not a generated model.

if you take a look at what `JsonCreator` annotation's description:
https://fasterxml.github.io/jackson-annotations/javadoc/2.7/com/fasterxml/jackson/annotation/JsonCreator.html

Speaking of what other SDK's behavior of including this annotation. They included it as well.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java [hunk 3]
```diff
@@ -144,43 +141,21 @@
...
-                    logger.logExceptionAsWarning(
-                        new RuntimeException(String.format(Locale.ROOT, "'%s' is not valid text sentiment.",
-                            sentenceSentiment.getSentiment())));
-                }
                 final SentimentConfidenceScorePerLabel confidenceScorePerSentence =
                     sentenceSentiment.getConfidenceScores();
-
                 return new SentenceSentiment(sentenceSentiment.getText(),
-                    sentenceSentimentLabel,
+                    sentenceSentiment.getSentiment().toString(),
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java:151]: We don't want to check for nulls anymore?
Initially, you were checking for null for `sentenceSentimentLabel` and `documentSentimentLabel` and throwing an exception.
Is it fine for sentimentLabel and documentSentimentLabel to be null?

Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java:151]: Yes. Same reason as https://github.com/Azure/azure-sdk-for-java/pull/11227/files/9ef54069c9a599fc03bf09ad1a957a4327023e31#r426102268

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/WarningCode.java (hunk 2 above)
Comment (samvaity) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/WarningCode.java:32]: It should be included on the deserialization layer i.e on the service side equivalent model. Don't need it here.

**Pull request merged**